src/hotspot/share/code/compressedStream.cpp

```diff
@@ -112,3 +112,106 @@ void CompressedWriteStream::write_long(jlong value) {
   write_signed_int(low(value));
   write_signed_int(high(value));
 }
+
+
+bool CompressedSparseDataReadStream::read_zero() {
```

Comment:
There is an issue here. If the last value written to a stream was 0, a reader would not know this is one 0 or eight 0s. Is there a guarantee that the number of reads will the same as the number of writes?

Reply:
Reader knows what he reads. We have no marks in a raw data.